source/MaterialXTest/GenShaderUtil.cpp

```diff
@@ -363,7 +336,8 @@ bool generateCode(mx::GenContext& context, const std::string& shaderName, mx::Ty
     bool stageFailed = false;
     for (auto stage : testStages)
     {
-        bool noSource = shader->getSourceCode(stage).empty();
+        sourceCode.push_back(shader->getSourceCode(stage));
+        bool noSource = sourceCode[sourceCode.size()-1].empty();
```

Comment:
Should you check earlier if sourceCode.size() is 0, cause if it is this will crash, no?

Reply:
I wouldn't worry about needing a test for push_back failing.. In that case we would need tests on std container allocation all over the code base ;)

But you could use `sourceCode.back()` instead of `sourceCode[sourceCode.size()-1]`.